### PR TITLE
Add Windows Support

### PR DESCRIPTION
**Pull request by Superloup10**

### CODE REVIEW

File: src/minecraft/java/search_jre.rs [hunk 1]
```diff
@@ -8,10 +8,16 @@ impl Java {
     #[cfg(windows)]
     pub fn runtime_path(&self) -> GrindstoneResult<PathBuf> {
         // Use MS store first if available
-
+        let mut path = PathBuf::from(dirs::data_local_dir().unwrap())
+            .join("Packages")
+            .join("Microsoft.4297127D64EC6_8wekyb3d8bbwe") // TODO: Find a way to get the package name
```
Comment (Bricklou) [inline src/minecraft/java/search_jre.rs]: Move this key to a constant

File: src/minecraft/java/search_jre.rs [hunk 2]
```diff
@@ -8,10 +8,16 @@ impl Java {
...
+            .join("LocalState");
+        path.push("runtime");
+        Ok(path)
+        //let mut path = dirs::data_local_dir().unwrap().as_path()
         // Otherwise, try the MSI installation version
 
-        unimplemented!("Not now, i need to setup a rust dev environment on one of my computers.");
+        //unimplemented!("Not now, i need to setup a rust dev environment on one of my computers.");
```
Comment (Bricklou) [inline src/minecraft/java/search_jre.rs]: don't forget to remove this line

**Review by Bricklou — CHANGES_REQUESTED** [review]

File: src/constants.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ pub const JAVA_JRE_MANIFEST_URL: &str = "https://launchermeta.mojang.com/v1/prod
 pub const MC_ASSETS_BASE_URL: &str = "https://resources.download.minecraft.net";
 
 #[cfg(windows)]
-/// Hash to detect Minecraft MS Store version folder
-pub const MC_MS_STORE_KEY: &str = "8wekyb3d8bbwe";
+pub const MC_MS_STORE_IDENTIFIANT: &str = "Microsoft.4297127D64EC6_8wekyb3d8bbwe";
```
Comment (Bricklou) [inline src/constants.rs]: In English ?

File: src/minecraft/java/download_file.rs [hunk 4]
```diff
@@ -109,7 +109,7 @@ impl Java {
 
             if path.exists() {
                 if !path.is_symlink() || path.read_link()? != target {
-                    std::fs::remove_file(&path)?;
+                    std::fs::remove_file(&path)?; // TODO Warning: Unnecessary qualification
```
Comment (Bricklou) [inline src/minecraft/java/download_file.rs]: What does "unnecessary qualification" stand for ?

Comment (Superloup10) [reply src/minecraft/java/download_file.rs]: It's `std::` which is not needed, `fs` is imported at the top of the file.

File: src/constants.rs (hunk 3 above)
Comment (Superloup10) [reply src/constants.rs]: My bad, `identifiant` sounded English to me.

**Review by Bricklou — APPROVED** [review]

**Pull request merged**